shared/rate_limits/__init__.py

```diff
@@ -1,17 +1,52 @@
 import logging
+from typing import Optional
 
 from redis import Redis, RedisError
 
-from shared.bots import get_adapter_auth_information
-from shared.bots.types import AdapterAuthInformation
 from shared.config import get_config
 from shared.django_apps.codecov_auth.models import Owner
 from shared.django_apps.core.models import Repository
+from shared.typings.torngit import TorngitInstanceData
 
 log = logging.getLogger(__name__)
 
 
-# Ensure service = "gh" before using
+def gh_app_key_name(installation_id: int, app_id: Optional[int] = None) -> str:
+    app_id = app_id or "default_app"
+    return f"{app_id}_{installation_id}"
+
+
+def owner_key_name(owner_id: int) -> str:
+    return str(owner_id)
+
+
+def bot_key_name() -> str:
+    return str(get_config("github", "bot", "key"))
+
+
+# TODO: Test this
+def determine_entity_redis_key_from_torngit_data(data: TorngitInstanceData) -> str:
+    """
+    Determines entity redis key from torngit data. This is an alternative
+    function to determine_entity_redis_key if the auth information has been
+    parsed into a torngit instance data object.
+
+    It should only be used for github git instances
+    """
+    current_installation = data.get("installation")
+    current_owner = data.get("owner")
+
+    if current_installation and current_installation.get("installation_id"):
+        return gh_app_key_name(
+            app_id=current_installation.get("app_id"),
+            installation_id=current_installation.get("installation_id"),
+        )
+    elif current_owner and current_owner.get("ownerid"):
+        return owner_key_name(owner_id=current_owner.get("ownerid"))
```

Comment:
I think there's an edge case here when the repo has a bot configured.

Notice that the AdapterAuthInfo includes the [token_owner](https://github.com/codecov/shared/blob/6ac0e7a2f571120969ccc0745c84444f45520ac4/shared/bots/__init__.py#L67), currently used for the token refresh functions.
The TorngitInstanceData on the other hand always uses [the repo owner](https://github.com/codecov/worker/blob/ed134ad2d06ab8f3a20738ee53e32762c41abeb2/services/repository.py#L108) as the owner on data.
This means that the token that is being used by the torngit instance might not actually belong to the owner in `data.get("owner")`, leading us to mark the `repo.owner` as rate limited when in fact is `repo.bot.owner` that has been rate limited

Reply:
Mmm so in this case, is would `token_owner` end up being `repo.bot.owner`?